lib/crystalball/graphql/map_generator/graphql_strategy/schema_execution_detector.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'graphql'
+
+module Crystalball
+  module GraphQL
+    module MapGenerator
+      class GraphQLStrategy
+        # Detects calls to `GraphQL::Schema.default_execution_strategy#execute`
+        # (default strategy is `GraphQL::Execution::Execute`) during example execution
+        # in order to add the type definition paths to the execution map
+        class SchemaExecutionDetector
+          class << self
+            attr_accessor :executed
+
+            # @return [Boolean] whether schema was executed
+            def schema_executed?
+              self.executed = false
+              old_strategy = ::GraphQL::Schema.default_execution_strategy
+              ::GraphQL::Schema.default_execution_strategy = Strategy
+              yield
+              executed
+            ensure
+              ::GraphQL::Schema.default_execution_strategy = old_strategy
+            end
+
+            # Simple GraphQL execution strategy that marks as executed and
+            # forwards to default implementation
+            class Strategy < ::GraphQL::Execution::Execute
+              def execute(*args)
+                SchemaExecutionDetector.executed = true
+                super(*args)
```

Comment:
Let's discuss it. I think GraphQL load types not only when you execute a query.

Reply:
Well, that's what [they say](https://github.com/rmosolgo/graphql-ruby/blob/952bb07367a810dbedf930b5133a19aba9ffd2f3/lib/graphql/define/instance_definable.rb#L12), and you can confirm it by breaking a `define` block, running a spec that does not execute `GraphQL` queries and seeing it pass.

Reply:
Change it to `ensure_defined`